example-generator/src/main/java/org/ehrbase/openehr/sdk/examplegenerator/ExampleGeneratorConfig.java

```diff
@@ -268,52 +272,143 @@ static void handleInstructionDetails(InstructionDetails value, WebTemplateNode n
-            Set<Transition> transitions = getInput(node.findChildById("transition").orElseThrow(), "code")
-                    .map(WebTemplateInput::getList)
-                    .stream()
-                    .flatMap(List::stream)
-                    .map(cs -> {
-                        String transitionName = cs.getValue();
-                        Transition transition = Arrays.stream(Transition.values())
-                                .filter(s -> s.getCode().equals(transitionName)).findFirst()
-                                .orElseThrow(() -> new IllegalArgumentException("Unknown transition: " + transitionName));
-                        return transition;
-                    })
-                    .collect(Collectors.toCollection(LinkedHashSet::new));
-
-            if (transitions.isEmpty()) {
-                transitions = EnumSet.allOf(Transition.class);
-                // filter transitions by permitted states
-                transitions.removeIf(t -> !states.contains(t.getTargetState()));
-            }
-
-            //take first transition
-            Transition transition = transitions.iterator().next();
-            State currentState = transition.getTargetState();
-
-            value.setCurrentState(currentState.toCodedText());
-            value.setTransition(transition.toCodedText());
+          Set<WebTemplateInputValue> theCurrentStates = getInput(node.findChildById("current_state").orElseThrow(), "code")
+              .map(WebTemplateInput::getList).stream()
+              .flatMap(List::stream)
+              .collect(Collectors.toSet());
+          
+          Set<State> allCurrentStates = convertToState(theCurrentStates);
+          
+          Set<WebTemplateInputValue> theTransitions = getInput(node.findChildById("transition").orElseThrow(), "code")
+              .map(WebTemplateInput::getList).stream()
+              .flatMap(List::stream)
+              .collect(Collectors.toCollection(LinkedHashSet::new));
+          
+          Set<Transition> allTransitions = convertToTransition(theTransitions);
+
+          Set<Pair<String,WebTemplateInputValue>> allCareflowTerminologyPairs = 
+              getInput(node.findChildById("careflow_step").orElseThrow(), "code")
+            .map(wti -> wti.getList().stream().map(wtiv -> Pair.of(wti.getTerminology(), wtiv)).collect(Collectors.toList()))
+            .stream()
+            .flatMap(l -> l.stream())
+            .collect(Collectors.toSet());
+          
+          //1. check if there is a match between careflow_step and transition related to a current_state
+          Optional<Triple<Pair<String, WebTemplateInputValue>,State,Transition>> candidate = allCurrentStates.stream()
+            .map(cs -> {
+              String code = cs.getCode();
+              Set<Pair<String,WebTemplateInputValue>> filteredCfs = filterCareflowSteps(allCareflowTerminologyPairs, code);
+              Set<Transition> filteredTrs = filterTransitions(allTransitions, code);
```

Comment:
`::filterTransitions` is only halled here. Why the detour State -> String -> State?